contracts/extensions/VotingBase.sol

```diff
@@ -0,0 +1,141 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "../../lib/dappsys/math.sol";
+import "../IColony.sol";
+import "../IColonyNetwork.sol";
+
+
+contract VotingBase is DSMath {
+
+  // Constants
+  uint256 constant REVEAL_PERIOD = 2 days;
+
+  // Initialization data
+  IColony colony;
+  IColonyNetwork colonyNetwork;
+
+  constructor(address _colony) public {
+    colony = IColony(_colony);
+    colonyNetwork = IColonyNetwork(colony.getColonyNetwork());
+  }
+
+  // Data structures
+  enum PollState { Pending, Open, Reveal, Closed, Executed }
+
+  struct Poll {
+    bool executed;
+    address creator;
+    uint256 pollCloses;
+    uint256[] voteCounts;
+  }
+
+  // Storage
+  uint256 pollCount;
+  mapping (uint256 => Poll) polls;
+  mapping (uint256 => mapping (uint256 => bytes)) pollActions;
+
+  // Modifiers
+  modifier onlyCreator(uint256 _pollId) {
+    require(polls[_pollId].creator == msg.sender, "voting-base-only-creator");
+    _;
+  }
+
+  modifier pollPending(uint256 _pollId) {
+    require(getPollState(_pollId) == PollState.Pending, "voting-base-poll-not-pending");
+    _;
+  }
+
+  // Functions
+  function createPoll() public {
+    pollCount += 1;
+    polls[pollCount].creator = msg.sender;
+  }
+
+  function addPollAction(uint256 _pollId, bytes memory _action) public onlyCreator(_pollId) pollPending(_pollId) {
```

Comment:
Working on the whitepaper, I certainly had in my head that votes would always be a binary choice - either this proposed thing would happen, or it wouldn't (this also resolves the tie issue, by saying for something to happen, it must have a clear majority). We can in principle have an unlimited number of choices here; I wonder if there are some subtle issues that arise in such a scenario. Not saying this is wrong, but we should think about it some more. For example, someone makes a poll that is:

* We should give Alice $10
* We should give Alice $11
* No action

These get 25%, 30% and 45% respectfully - clearly a majority for giving Alice at least $10, but instead she gets nothing. The counterargument is that two polls could be made, one for giving Alice $10 or nothing, and one for giving Alice $11 or nothing, and Alice might end up with $21 - but that could happen here as well.

So not saying this is wrong, but we should think about it some more (possibly even bouncing it back to a research context).

Reply:
That's a fair point. In my mind implementing multi-option voting is more general (i.e. we can always create a `createBinaryPoll` function which under-the-hood creates a two-option poll in one go), but I would be happy limiting things to two options -- it would make poll creation much more straightforward.

The situation I would imagine for multiple options would be an election (i.e. vote on the domain administrator), but we could make a parallel set of `Election` extensions for that purpose.

Reply:
As an aside, in cases where we're deciding on some numerical value, I think a contract which implements some type of range voting / aggregation which can operate on real numbers is absolutely the way to go (i.e. everyone submits their number and we output the median).